pyaedt/edb_core/edb_data/simulation_setup.py

```diff
@@ -0,0 +1,701 @@
+
+    @pyaedt_function_handler()
+    def _update_sweep(self):
+        """Update sweep."""
+        self._sim_setup._edb_object.SweepDataList.Clear()
+        for el in list(self._sim_setup.frequency_sweeps.values()):
+            self._sim_setup._edb_object.SweepDataList.Add(el._edb_sweep_data)
+        self._sim_setup._edb_object.SweepDataList.Add(self._edb_sweep_data)
+        return self._sim_setup._update_setup()
+
+    @property
+    def name(self):
+        """Name of the sweep."""
+        return self._edb_sweep_data.Name
+
+    @name.setter
+    def name(self, value):
+        """Set name of this sweep"""
+        self._edb_sweep_data.Name = value
+        self._update_sweep()
+
+    @property
+    def sweep_type(self):
+        """Sweep type."""
+        return
+
+    @property
+    def frequencies(self):
+        """List of frequencies points."""
+        return list(self._edb_sweep_data.Frequencies)
+
+    @property
+    def adaptive_sampling(self):
+        """Whether adaptive sampling is used.
+
+        Returns
+        -------
+        bool
+            ``True`` if adaptive sampling is used, ``False`` otherwise.
+        """
+        return self._edb_sweep_data.AdaptiveSampling
+
+    @property
+    def adv_dc_extrapolation(self):
+        """Whether to turn on advanced DC Extrapolation.
+
+        Returns
+        -------
+        bool
+            ``True`` if advanced DC Extrapolation is used, ``False`` otherwise.
+
+        """
+        return self._edb_sweep_data.AdvDCExtrapolation
+
+    @property
+    def auto_s_mat_only_solve(self):
+        """Whether to turn on Auto/Manual SMatrix only solve.
+
+        Returns
+        -------
+        bool
+            ``True`` if Auto/Manual SMatrix only solve is used, ``False`` otherwise.
+        """
+        return self._edb_sweep_data.AutoSMatOnlySolve
+
+    @property
+    def enforce_causality(self):
+        """Whether to enforce causality during interpolating sweep.
+
+        Returns
+        -------
+        bool
+            ``True`` if enforce causality is used, ``False`` otherwise.
+        """
+        return self._edb_sweep_data.EnforceCausality
+
+    @property
+    def enforce_dc_and_causality(self):
+        """Whether to enforce DC point and causality.
+
+        Returns
+        -------
+        bool
+            ``True`` if enforce dc point and causality is used, ``False`` otherwise.
+
+        """
+        return self._edb_sweep_data.EnforceDCAndCausality
+
+    @property
+    def enforce_passivity(self):
+        """Whether to enforce passivity during interpolating sweep.
+
+        Returns
+        -------
+        bool
+            ``True`` if enforce passivity is used, ``False`` otherwise.
+        """
+        return self._edb_sweep_data.EnforcePassivity
+
+    @property
+    def freq_sweep_type(self):
+        """Sweep type.
+        Options are:
+        - ``kInterpolatingSweep``.
+        - ``kDiscreteSweep``.
+        - ``kBroadbandFastSweep``.
```

Comment:
```suggestion
        """Sweep type. Options are:

        - ``kInterpolatingSweep``.
        - ``kDiscreteSweep``.
        - ``kBroadbandFastSweep``.
```
To render correctly in the doc, you must have a blank line before the first item in a list and after the last item in a list.